model/map/src/main/java/org/keycloak/models/map/events/MapEventStoreProvider.java

```diff
@@ -179,4 +175,11 @@ public void clearAdmin(RealmModel realm, long olderThan) {
     public void close() {
 
     }
+    @Override
```

Comment:
Expiration works differently in the new storage. 

Previously the event expiration duration was used at the time of removing events. The scheduled task executed a query that removed all entities with a time (time of firing the event) is less than something.

In the new store, we compute the expiration timestamp at the event firing time. This way the cleanup task does not depend on any information from the realm and cleanup can be done independently of Keycloak.

Therefore we need to compute this expiration time in this file. You can see how we do it for auth events:
https://github.com/keycloak/keycloak/blob/3d2c3fbc6aa73116edeaabf6b3a7f8ba0ae9769c/model/map/src/main/java/org/keycloak/models/map/events/MapEventStoreProvider.java#L71-L79

Reply:
Hi, I saw that in the comments and was already thinking of adding an expiration timestamp to the events. 
This would also make the cleanup query easier and probably faster. I did not go that way, because it would mean a db migration which was (in another context) rejected in this PR.

Reply:
The expiration field is already present in the new store [MapAdminEventEntity](https://github.com/keycloak/keycloak/blob/main/model/map/src/main/java/org/keycloak/models/map/events/MapAdminEventEntity.java) (see `ExpirableEntity` interface). DB migration is not necessary at the moment, because we added the expiration field only to the new store, the legacy store will continue working as now.

Basically, what you need is to add the snippet of code I sent in the first comment to [onEvent method](https://github.com/keycloak/keycloak/blob/main/model/map/src/main/java/org/keycloak/models/map/events/MapEventStoreProvider.java#L132).

Reply:
Ah yes, sorry I was not clear on this. I was thinking about adding the expiration field also to the Jpa(Admin)EventEntity. I am not yet working on the Map stuff.